src/main/java/duke/AddDeadlineCommand.java

```diff
@@ -0,0 +1,92 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.format.FormatStyle;
+
+/**
+ * Encapsulates a Add Deadline Command.
+ */
+public class AddDeadlineCommand extends Command {
+    /**
+     * Instantiates a AddDeadlineCommand.
+     * @param parsedCommand the parsed command
+     */
+    public AddDeadlineCommand(String[] parsedCommand) {
+        super(parsedCommand);
+    }
+
+    /**
+     * Executes command and write to storage.
+     * @param tasks the tasklist containing tasks so far
+     * @param ui ui to interact with user
+     * @param storage storage to read and write to storage file
+     * @throws DukeException if parsedCommand does not meet the requirements
+     */
+    @Override
+    void execute(TaskList<Task> tasks, Ui ui, Storage storage) throws DukeException {
+        // if length is not 2, nothing was passed in after 'deadline'
+        if (getParsedCommand().length != 2) {
+            throw new DukeException("NANI??! Enter a description for your deadline!\n");
+        }
+
+        // if description is lacking a /by keyword
+        String description = getParsedCommand()[1];
+        if (description.indexOf("/by") < 0) {
+            throw new DukeException("Please enter a valid deadline! Remember to add '/by'\n");
+        }
+
+        String[] descriptionArray = description.split("/by");
+        if (descriptionArray.length != 2) {
+            throw new DukeException("NANI??! Enter your deadline name, date & time(if any) (e.g. 2020-01-30 18:30)!\n");
+        }
+
```

Comment:
This execute function is quite long. Maybe can break up further into parts for better readability?

Reply:
Thank you for your constructive comment! I will get to it.